### PR TITLE
Add the Expo Offline support page

### PR DESCRIPTION
**Pull request by anagstef**
> [!IMPORTANT]
> 🔎 Previews:
>
> - https://clerk.com/docs/pr/1781/references/expo/offline-support

### Explanation:

- We introduce a new experimental feature that allows Clerk to cache its resources on Expo, so it can be initialized when the network is not online. 
- ✅  **This page should not go live until the https://github.com/clerk/javascript/pull/4604 is merged and released!**

### This PR:

- Adds a new page that adds documentation for Expo offline support experimental feature.

### Checklist

- [x] I have clicked on "Files changed" and performed a thorough self-review
- [x] I have added the "deploy-preview" label and added the preview link(s) to this PR description
- [x] All existing checks pass

### CODE REVIEW

**Comment by alexisintech** [comment]
posted my code review - let me know if it looks okay :)

**Comment by anagstef** [comment]
@alexisintech Looks great! Thank you! 

Can we also mark line 4?
`import { secureStore } from '@clerk/clerk-expo/secure-store'`

File: docs/references/expo/offline-support.mdx [hunk 1]
```diff
@@ -0,0 +1,72 @@
+---
+title: Enable offline support in your Expo app
+description: Learn how to enable offline support in your Expo app with Clerk.
+---
+
+> [!WARNING]
+> **This is an experimental API.**
+>
+> The `__experimental_resourceCache` property introduced in this guide is an experimental feature. It is subject to change in future updates, so use it cautiously in production environments. Ensure thorough testing and stay informed about updates in the package.
```
Comment (royanger) [inline docs/references/expo/offline-support.mdx:9]: Can we share where/how to stay informed here?

**Review by royanger — APPROVED** [review]
Glad to see this getting added (docs and the support both)!

File: docs/references/expo/offline-support.mdx (hunk 1 above)
Comment (anagstef) [reply docs/references/expo/offline-support.mdx:9]: Hmm, should we point them to the changelog of our Expo package? 🤔

Comment (royanger) [reply docs/references/expo/offline-support.mdx:9]: If that's the spot where updates will be published, then I think that's the best choice.

Comment (anagstef) [reply docs/references/expo/offline-support.mdx:9]: Yes, this is the only spot I believe

**Review by alexisintech — APPROVED** [review]
thanks for getting this in there!! looks great 😸💖

**Comment by alexisintech** [comment]
is offline support for expo web or expo native or both? it looks like the manifest organization got messed up

**Comment by alexisintech** [comment]
waiting to see if the hooks are for native, web, or both

**Comment by anagstef** [comment]
@alexisintech Both the hooks and the offline support are used only on the Expo native.

File: docs/references/expo/offline-support.mdx (hunk 1 above)
Comment (anagstef) [reply docs/references/expo/offline-support.mdx]: Added the link to the changelog!

**Review by alexisintech — APPROVED** [review]
woohoo 😸💖

**Pull request merged**